### PR TITLE
Add `Dockerfile`, `docker-compose.yml` and `heroku.yml` to use Docker image as build artifact in heroku

### PR DESCRIPTION
**Pull request by Staberinde**
* Added `Dockerfile` with `dev` and `live` targets
* Added `docker-compose.yml` and enabled live reloading of python code
* Added `heroku.yml` and set it up to push `live` image to heroku

# TODO
* [x] Figure out how database migrations are currently done with heroku buildpack and implement

### CODE REVIEW

File: requirements/dev-requirements.txt [hunk 1]
```diff
@@ -1,5 +1,5 @@
 #
-# This file is autogenerated by pip-compile with python 3.9
+# This file is autogenerated by pip-compile with python 3.8
```
Comment (acodeninja) [inline requirements/dev-requirements.txt:2]: Does it make a difference that we are generating this with python 3.8 but the docker container is running python 3.10?

File: setup.py [hunk 2]
```diff
@@ -0,0 +1,15 @@
+from setuptools import find_packages, setup
+
+from version import get_version
+
+setup(
+    name="digital-land-data-manager",
+    version=get_version(),
+    author="MHCLG Digital Land Team",
```
Comment (acodeninja) [inline setup.py:8]: This should probably read DLUHC.

File: setup.py [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+    author_email="DigitalLand@communities.gov.uk",
+    license="MIT",
+    url="https://github.com/digital-land/pipeline",
```
Comment (acodeninja) [inline setup.py:11]: This should probably be the URL of this repo rather than an old link to `digital-land-python` repo.

File: setup.py (hunk 2 above)
Comment (Staberinde) [reply setup.py:8]: :+1: I copied a lot of this file verbatim from `digital-land-python`

**Review by acodeninja — APPROVED** [review]

**Pull request merged**